tests/test_api.py

```diff
@@ -107,6 +110,61 @@ def test_api_transmute_tarball_to_conda_v2(testing_workdir):
     assert os.path.isfile(os.path.join(testing_workdir, test_package_name + '.conda'))
 
 
+@pytest.mark.skipif(sys.platform=="win32", reason="windows and symlinks are not great")
+def test_api_transmute_to_conda_v2_contents(testing_workdir):
+    def _walk(path):
+        for entry in os.scandir(path):
+            if entry.is_dir(follow_symlinks=False):
+                yield from _walk(entry.path)
+                continue
+            yield entry
+
+    tar_path = os.path.join(data_dir, test_package_name_2 + '.tar.bz2')
+    conda_path = os.path.join(testing_workdir, test_package_name_2 + '.conda')
+    api.transmute(tar_path, '.conda', testing_workdir)
+
+    # Verify original contents were all put in the right place
+    pkg_tarbz2 = tarfile.open(tar_path, mode="r:bz2")
+    info_items = [item for item in pkg_tarbz2.getmembers()
+                  if item.path.startswith("info/")]
+    pkg_items = [item for item in pkg_tarbz2.getmembers()
+                 if not item.path.startswith("info/")]
+
+    errors = []
+    for component, expected in (("info", info_items), ("pkg", pkg_items)):
+        with TemporaryDirectory() as root:
+            api.extract(conda_path, root, components=component)
+
+            contents = {
+                os.path.relpath(entry.path, root): {
+                    "is_symlink": entry.is_symlink(),
+                    "target": os.readlink(entry.path) if entry.is_symlink() else None
+                    }
+                for entry in _walk(root)
+                }
+
+            for e in expected:
+                if e.path not in contents:
+                    errors.append(f"'{e.path}' not found in {component} contents")
+                    continue
+
+                ct = contents.pop(e.path)
+                if e.issym():
+                    if not ct["is_symlink"] or ct["target"] != e.linkname:
+                        errors.append(f"{e.name} -> {e.linkname} incorrect in {component} contents")
+                elif not e.isfile():
+                    # Raise an exception rather than appending to `errors`
+                    # because getting to this point is an indication that our
+                    # test data (i.e., .tar.bz2 package) is corrupt, rather
+                    # than the `.transmute` function having problems (which is
+                    # what `errors` is meant to track).  For context, conda
+                    # packages should only contain regular files and symlinks.
+                    raise ValueError(f"unexpected item '{e.path}' in test .tar.bz2")
```

Comment:
```suggestion
            for item in expected:
                if item.path not in contents:
                    errors.append(f"'{item.path}' not found in {component} contents")
                    continue

                ct = contents.pop(item.path)
                if item.issym():
                    if not ct["is_symlink"] or ct["target"] != item.linkname:
                        errors.append(f"{item.name} -> {item.linkname} incorrect in {component} contents")
                elif not item.isfile():
                    # Raise an exception rather than appending to `errors`
                    # because getting to this point is an indication that our
                    # test data (i.e., .tar.bz2 package) is corrupt, rather
                    # than the `.transmute` function having problems (which is
                    # what `errors` is meant to track).  For context, conda
                    # packages should only contain regular files and symlinks.
                    raise ValueError(f"unexpected item '{item.path}' in test .tar.bz2")
```

Just a nit..

Reply:
It's an important nit though. :)